imap_processing/swe/l2/swe_l2.py

```diff
@@ -195,6 +201,59 @@ def calculate_flux(l1b_dataset: xr.Dataset) -> npt.NDArray:
     return flux
 
 
+def put_data_into_angle_bins(
+    data: np.ndarray, angle_bin_indices: npt.NDArray[np.int_]
+) -> npt.NDArray:
+    """
+    Put data in its angle bins.
+
+    This function bins SWE data into 30 predefined angle bins
+    while preserving the original energy step structure. For each
+    full cycle, it assigns data to the corresponding angle bin
+    based on the provided indices.
+
+    Since multiple data points may fall into the same angle bin,
+    the function accumulates values and computes the average across
+    all 7 CEMs, ensuring that each bin contains a representative
+    mean value while maintaining the 7 CEM structure.
+
+    Parameters
+    ----------
+    data : numpy.ndarray
+        Data to put in bins. Shape: (full_cycle_data, energy_step, angle_bin, 7).
+    angle_bin_indices : numpy.ndarray
+        Indices of angle bins to put data in. Shape:
+        (full_cycle_data, energy_step, angle_bin).
+
+    Returns
+    -------
+    numpy.ndarray
+        Data in bins. Shape: (full_cycle_data, 24, 30, 7).
+    """
+    # Initialize with zeros instead of NaN because np.add.at() does not
+    # work with nan values. It results in nan + value = nan
+    binned_data = np.zeros((data.shape[0], 24, 30, 7), dtype=np.float64)
+
+    time_indices = np.arange(data.shape[0])[:, None, None]
+    energy_indices = np.arange(data.shape[1])[None, :, None]
```

Comment:
For the future, I wonder if we should make these global constants as well? You use the 24, 30, 7 values a lot throughout the code. `N_ESA_STEPS`, `N_ANGLES`, `N_CEMS`
```suggestion
    energy_indices = np.arange(24)[None, :, None]
```

Reply:
sounds good. I can do that in future PR